packages/grid/x-data-grid/src/hooks/features/export/serializers/csvSerializer.ts

```diff
@@ -48,6 +49,43 @@ const objectFormattedValueWarning = buildWarning([
   'You can provide a `valueFormatter` with a string representation to be used.',
 ]);
 
+type CSVRowOptions = {
+  delimiterCharacter: string;
+  sanitizeValue?: boolean;
+};
+class CSVRow {
+  options: CSVRowOptions;
+
+  rowString = '';
+
+  isEmpty = true;
+
+  constructor(options: CSVRowOptions) {
+    this.options = options;
+    if (this.options.sanitizeValue === undefined) {
+      this.options.sanitizeValue = true;
+    }
+  }
+
+  addValue(value: string) {
+    if (!this.isEmpty) {
+      this.rowString += this.options.delimiterCharacter;
+    }
+    if (value === null || value === undefined) {
+      this.rowString += '';
+    } else if (this.options.sanitizeValue) {
```

Comment:
Instead of boolean, probably pass the `sanitizeCellValue` function whenever sanitization is needed to make this class a bit pure?
```suggestion
    } else if (this.options.sanitizeCellValue) {
```